### PR TITLE
Supporting RBD persistent write bach cache feature basic functionality

### PR DESCRIPTION
**Pull request by sunilkumarn417**
- Added basic functionality validation of RBD persistent write back(pwl) chache.
  - Required libraries.
  - Test cases which covers basic validation pwl at client, image and pool level.
  - Fix OSD validation issue, which was happening due to available and unavailable devices on refresh.
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-OCCN99/deploy_cluster_0.log

**Test cases:**
- [CEPH-83574721](https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83574721) check Configurations rbd_persistent_cache_path in SSD mode
   - Configure PWL at client level with default size 1 GB.
   - Configure cache client with DAX option.
   - run FIO and paralelly, validate cache file and its path of the cache from RBS status command.
   - Repeat the above steps for pool and image level.
- [CEPH-83574707](https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83574707) check the cache creation in the rbd_persistent_cache_path in SSD mode 
  - Configure PWL at client level with default size 1 GB.
  - Configure cache client with DAX option.
  - run FIO and paralelly validate file of the cache from RBD status command.
  - repeat above steps for pool and image level.
- [CEPH-83574722](https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83574722) check Configurations rbd_persistent_cache_size in SSD mode 
   - Configure PWL at client level with default size 1 GB.
   - Configure cache client with DAX option.
   - run FIO and paralelly validate file, size of the cache from RBD status command.
   - repeat above steps for pool and image level.

**Results:** 
(RHCS 6.x)
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-CSKU01 ( Passed -old)
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-DJD6W8  
(RHCS 5.x) 
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-YG72N5
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-SJQUDC
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-VVV44N/ ( latest )

### CODE REVIEW

File: tests/rbd/rbd_utils.py [hunk 1]
```diff
@@ -396,6 +397,21 @@ def list_images(self, pool_name):
...
         return images
 
+    def image_status(self, image, **kw):
+        """Get Image status.
+
+        Command: rbd status <pool-name>/<image-name>
+
+        Args:
+            image: name of the image (ex., pool1/image1)
+            kw: other test parameters ( ex., {output: True, json: true})
```
Comment (VasishtaShastry) [inline tests/rbd/rbd_utils.py:407]: [Optional]

Change image to image specification / image spec
image could be confused with actual image name, mentioning as optional as example will help to understand
Usually `pool/image` is called image specification

Comment (VasishtaShastry) [inline tests/rbd/rbd_utils.py:407]: [Optional]

Change image to image specification / image spec
image could be confused with actual image name, mentioning as optional as example will help to understand
Usually `pool/image` is called image specification

Comment (VasishtaShastry) [reply tests/rbd/rbd_utils.py:407]: (Duplicate comment)

Comment (sunilkumarn417) [reply tests/rbd/rbd_utils.py:407]: Provided in the command above. But it make sense to call it as image_spec.
Let me apply that.

Comment (sunilkumarn417) [reply tests/rbd/rbd_utils.py]: Done. updated.

Results:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-87DEHE

File: conf/quincy/upi/octo-5-node-env.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    ceph-cluster:
+      name: ceph
+      networks:
+        public: ['10.8.128.0/21']
+      nodes:
+        -
+          hostname: clara003
+          id: node1
+          ip: 10.8.129.3
+          root_password: passwd
```
Comment (sunilangadi2) [inline conf/quincy/upi/octo-5-node-env.yaml:17]: Wouldn't it cause a `security threat` as we are passing and hard coding root password here,
Can we make it self configurable from user in `cephci.yaml` template?

File: suites/quincy/rbd/rbd_persistent_write_back_cache.yaml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+              size: 20G
+            fio:
+              image_name: image2
+              pool_name: pool2
+              runtime: 120
+          desc: PWL validation at pool level
+          destroy-cluster: false
+          module: rbd_persistent_write_back_cache.py
+          name: RBD Persistent Cache - Pool level configuration
+          polarion-id:
```
Comment (sunilangadi2) [inline suites/quincy/rbd/rbd_persistent_write_back_cache.yaml:125]: polarion id is missing, please add that

File: suites/quincy/rbd/rbd_persistent_write_back_cache.yaml [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+              image: image3
+              size: 40G
+            fio:
+              image_name: image3
+              pool_name: pool3
+              runtime: 120
+          desc: PWL validation at image level
+          module: rbd_persistent_write_back_cache.py
+          name: RBD Persistent Cache - image level configuration
+          polarion-id:
```
Comment (sunilangadi2) [inline suites/quincy/rbd/rbd_persistent_write_back_cache.yaml:146]: same here

**Review by sunilangadi2 — COMMENTED** [review]
Can we cover this test in pacific as well?

File: suites/quincy/rbd/rbd_persistent_write_back_cache.yaml (hunk 3 above)
Comment (sunilkumarn417) [reply suites/quincy/rbd/rbd_persistent_write_back_cache.yaml:125]: Oh, need to take that away. the actual test case polarion id is added at parallel.

File: conf/quincy/upi/octo-5-node-env.yaml (hunk 2 above)
Comment (sunilkumarn417) [reply conf/quincy/upi/octo-5-node-env.yaml:17]: Currently, all UPI tests are configured this way, until we have new provisioning.
please check this file, https://github.com/red-hat-storage/cephci/blob/master/conf/quincy/upi/octo_11_node_env.yaml

File: suites/quincy/rbd/rbd_persistent_write_back_cache.yaml [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+      desc: Configure client node
+      destroy-cluster: false
+      module: test_client.py
+      name: configure client
+      polarion-id: CEPH-83573758
+
+  - test:
+      abort-on-fail: true
+      config:
+        level: global                        # PWL at global
```
Comment (preethisn) [inline suites/quincy/rbd/rbd_persistent_write_back_cache.yaml:76]: @sunilkumarn417 We dont have such as global level for SSD- Its client level configuration. Please change accordingly

**Comment by sunilkumarn417** [comment]
> Can we cover this test in pacific as well?
Yes, Supported in pacific!!  http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-YG72N5

File: suites/quincy/rbd/rbd_persistent_write_back_cache.yaml [hunk 6]
```diff
@@ -0,0 +1,146 @@
+# RBD: Persistent write back cache feature
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply suites/quincy/rbd/rbd_persistent_write_back_cache.yaml:1]: ok

File: suites/quincy/rbd/rbd_persistent_write_back_cache.yaml (hunk 5 above)
Comment (sunilkumarn417) [reply suites/quincy/rbd/rbd_persistent_write_back_cache.yaml]: Done

**Review by manasagowri — CHANGES_REQUESTED** [review]

File: suites/quincy/rbd/tier-3_rbd_persistent_write_back_cache.yaml [hunk 7]
```diff
@@ -0,0 +1,144 @@
+# RBD: Persistent write back cache feature
```
Comment (sunilangadi2) [inline suites/quincy/rbd/tier-3_rbd_persistent_write_back_cache.yaml:1]: Add this suite in Pacific as well along with the pipeline metadata file.

File: conf/quincy/upi/octo-5-node-env.yaml [hunk 8]
```diff
@@ -0,0 +1,88 @@
+# Physical/Bare-metal environment required for RBD persistent write back cache.
```
Comment (sunilangadi2) [inline conf/quincy/upi/octo-5-node-env.yaml:1]: If needed Add this conf in pacific as well.

Comment (sunilkumarn417) [reply conf/quincy/upi/octo-5-node-env.yaml:1]: @sunilangadi2 , This is next step after this PR merge. And for this activity, need Amar's help.

File: suites/quincy/rbd/tier-3_rbd_persistent_write_back_cache.yaml (hunk 7 above)
Comment (sunilkumarn417) [reply suites/quincy/rbd/tier-3_rbd_persistent_write_back_cache.yaml:1]: @sunilangadi2 , This is next step after this PR merge. And for this activity, need Amar's help.

**Review by sunilangadi2 — DISMISSED** [review]
My requested changes look good to me

**Comment by sunilkumarn417** [comment]
Latest results: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-35IZBO/

**Review by manasagowri — APPROVED** [review]
Group review done. Changes LGTM

**Pull request merged**